evalml/tests/component_tests/test_polynomial_detrender.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+import pandas as pd
+import pytest
+import woodwork as ww
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import PolynomialFeatures
+
+from evalml.pipelines.components import PolynomialDetrender
+
+pytest.importorskip('sktime', reason='Skipping polynomial detrending tests because sktime not installed')
+
+
+def test_polynomial_detrender_init():
+    delayed_features = PolynomialDetrender(degree=3)
+    assert delayed_features.parameters == {"degree": 3}
+
+
+def test_polynomial_detrender_raises_value_error_target_is_none(ts_data):
+    X, y = ts_data
+
+    with pytest.raises(ValueError, match="y cannot be None for PolynomialDetrender!"):
+        PolynomialDetrender(degree=3).fit_transform(X, None)
+
+    pdt = PolynomialDetrender(degree=3).fit(X, y)
+    with pytest.raises(ValueError, match="y cannot be None for PolynomialDetrender!"):
+        pdt.inverse_transform(X, None)
+
+
+@pytest.mark.parametrize("input_type", ['np', 'pd', 'ww'])
+@pytest.mark.parametrize("use_int_index", [True, False])
+@pytest.mark.parametrize("degree", [1, 2, 3])
+def test_polynomial_detrender_fit_transform(degree, use_int_index, input_type, ts_data):
+
+    X_input, y_input = ts_data
+
+    if use_int_index:
+        X_input.index = np.arange(X_input.shape[0])
+        y_input.index = np.arange(y_input.shape[0])
+
+    # Get the expected answer
+    lin_reg = LinearRegression(fit_intercept=True)
+    features = PolynomialFeatures(degree=degree).fit_transform(np.arange(X_input.shape[0]).reshape(-1, 1))
+    lin_reg.fit(features, y_input)
+    detrended_values = y_input.values - lin_reg.predict(features)
+    expected_index = y_input.index if input_type != 'np' else range(y_input.shape[0])
+    expected_answer = pd.Series(detrended_values, index=expected_index)
+
+    X, y = X_input, y_input
+
+    if input_type == 'np':
+        X = X_input.values
+        y = y_input.values
+    elif input_type == 'ww':
+        X = ww.DataTable(X_input)
+        y = ww.DataColumn(y_input)
```

Comment:
We can use the `make_data_type` fixture for this I believe? hehe

Reply:
Thanks for the tip! I didn't know about `make_data_type`! I'm not sure if it's worth it in this case because there is no pandas to numpy conversion and the woodwork conversion is straight-forward enough in this case. But I will keep it in mind in the future.